### PR TITLE
Milestone: processing data in models

### PR DESCRIPTION
**Pull request by sanieni6**
# Blog App - processing data in models 🤓 
In this part of the project I created models for each table, relations and custom methods
## Summary

- Create model classes for all entities ✅ 
- Set up associations between models ✅ 
- Use models to insert data ✅ 
- Create custom methods ✅

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (Ismailco) [inline app/models/user.rb]: - [ ] [OPTIONAL] Kindly consider adding `foreign_keys` to the has_many relationships 

```suggestion
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (Ismailco) [inline app/models/post.rb]: - [ ] The post belongs to the author using the user class name, kindly make sure to change the user to the author
e.g
```
belongs_to :author, class_name: 'User' 
```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (Ismailco) [inline app/models/like.rb]: - [ ] The like belongs to the author using the user class name, kindly make sure to change the user to the author

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (Ismailco) [inline app/models/comment.rb]: - [ ] The comment belongs to the author using the user class name, kindly make sure to change the user to the author

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+  def update_post_counter
+    user = User.find_by(id: author_id)
+    return unless user
+
+    user.update(posts_counter: Post.where(author_id:).count)
+  end
+
+  def most_recent_comments
+    Comment.where(post_id: id).order(created_at: :desc)
```
Comment (Ismailco) [inline app/models/post.rb]: - [ ] This method should list only the last 5 comments, kindly make sure to fix this so you can list the 5 most recent comments

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @sanieni6, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Create Models
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Whoistolu) [reply app/models/user.rb]: - Well resolved 👍🏽

File: app/models/post.rb (hunk 2 above)
Comment (Whoistolu) [reply app/models/post.rb]: - Well resolved 👍🏽

File: app/models/like.rb (hunk 3 above)
Comment (Whoistolu) [reply app/models/like.rb]: - Well resolved 👍🏽

File: app/models/comment.rb (hunk 4 above)
Comment (Whoistolu) [reply app/models/comment.rb]: - Well resolved 👍🏽

File: app/models/post.rb (hunk 5 above)
Comment (Whoistolu) [reply app/models/post.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @sanieni6 

Great job implementing the recommended changes by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**